lib/client.js

```diff
@@ -1208,7 +1340,7 @@ function _resume (client, sync) {
 
     const socket = client[kSocket]
 
-    if (socket && !socket.destroyed) {
+    if (socket && !socket.destroyed && socket.alpnProtocol !== 'h2') {
```

Comment:
Don't we need to ref/unref the h2 sockets? Is it harmful if we do ref/unref?

Reply:
We are using the same socket for the `h2` connection, and then we can follow the same pattern as per `h1`.
It should not be a problem, as no more clients running is the criteria; no more traffic is flowing although the streams are still open.
As we are not supporting `push` nor `ping` or similar, this should not be a problem, once we decide to introduce it we we'll need to check this once more.